hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkGetAccountDetailsHandler.java

```diff
@@ -196,42 +200,60 @@ private static Optional<AccountDetails> infoForAccount(
      * @param tokenRelationStore the token relationship store
      * @return list of token relationships for the given account
      */
-    private static List<TokenRelationship> getTokenRelationships(
+    private List<TokenRelationship> tokenRelationshipsOf(
             final long maxRelsPerInfoQuery,
-            Account account,
-            ReadableTokenStore readableTokenStore,
-            ReadableTokenRelationStore tokenRelationStore) {
-        final var tokenRelationshipList = new ArrayList<TokenRelationship>();
+            @NonNull final Account account,
+            @NonNull final ReadableTokenStore readableTokenStore,
+            @NonNull final ReadableTokenRelationStore tokenRelationStore) {
+        requireNonNull(account);
+        requireNonNull(tokenRelationStore);
+        requireNonNull(readableTokenStore);
+
+        final var ret = new ArrayList<TokenRelationship>();
         var tokenId = account.headTokenId();
         int count = 0;
-
-        while (tokenId != null && !tokenId.equals(TokenID.DEFAULT) && count <= maxRelsPerInfoQuery) {
-            final var tokenRelation = tokenRelationStore.get(account.accountId(), tokenId);
+        TokenRelation tokenRelation;
+        Token token; // token from readableToken store by tokenID
+        AccountID accountID; // build from accountNumber
+        while (tokenId != null && !tokenId.equals(TokenID.DEFAULT) && count < maxRelsPerInfoQuery) {
```

Comment:
I'm uneasy about patterns where we treat it as possible for `TokenID.DEFAULT` to be used as an alternative to `null` to indicate a missing field.

Reply:
Removed that. Thanks @tinker-michaelj